src/ItemFieldSet.tsx

```diff
@@ -16,59 +32,215 @@ interface IItemFieldSetProps {
+        }
+        label={name}
+      />
+      )
     });
-    return this.state.items.map((item: IItem, index: number) => {
-      return (
-        <Card key={index}>
+    const splitBillButton = (<Button id="splitBtn" variant="contained" color="primary" onClick={this.setItems} disabled={!this.isValid()}>
+      Split bill
+		</Button>);
+    const addButton = (<IconButton style={{ float: 'right' }} iaria-label="Add " onClick={this.add}>
+      <AddCircleIcon color="secondary" />
+    </IconButton>);
+    const itemsArray = this.state.items.map((item: IItemInternal, itemIndex: number) => {
+      return (<div key={itemIndex}>
+        <Card key={itemIndex}>
           <CardContent>
             <p>Describe Item Consumed</p>
             <div>
               <TextField type="text"
                 label="Product Name"
                 value={item.name}
-                onChange={this.changeItemName(index)}
+                onChange={this.changeItemName(itemIndex)}
+              />
+            </div>
+            <br />
+            <div>
+              <FormLabel component="legend">Quantity</FormLabel>
+              <TextField type="text"
+                value={this.getItemQuantityString(itemIndex)}
+                onChange={this.changeItemQuantity(itemIndex)}
+                onFocus={this.itemQuantityToggleFocus(itemIndex, true)}
+                onBlur={this.itemQuantityToggleFocus(itemIndex, false)}
               />
             </div>
+            <br />
             <div>
+              <FormLabel component="legend">Price</FormLabel>
               <TextField type="text"
-                label="Quantity"
-                value={item.quantity}
-                onChange={this.changeItemQuantity(index)}
+                value={this.moneyDecimalToString(item.price, item.isFocused)}
+                onChange={this.changeItemPrice(itemIndex)}
+                onFocus={this.itemPriceToggleFocus(itemIndex, true)}
+                onBlur={this.itemPriceToggleFocus(itemIndex, false)}
               />
             </div>
-            {names}
+            <br />
+            <div>
+              <FormLabel component="legend">Who consumed</FormLabel>
+              <FormGroup>
+                {names(itemIndex)}
+              </FormGroup>
+            </div>
           </CardContent>
         </Card>
+      </div>
       );
     });
+    return (<div>{itemsArray}{splitBillButton}{addButton}</div>)
+  }
+
+  private add = () => {
+    const newItem: IItemInternal = { isFocused: false, name: "", payingIndexes: [], quantity: 1, price: "" }
+    const items: IItemInternal[] = [...this.state.items, newItem];
+    this.setState({ items });
+  }
+
+  private getItemQuantityString = (index: number, isFocused?: boolean) => {
+    if (this.state.items[index].isFocused && isNaN(this.state.items[index].quantity)) {
```

Comment:
you repeat this.state.items[index].quantity 3 times... you can use const quantity = this.state.items[index].quantity; and then just use quantity.